Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -2149,7 +2150,7 @@ static void fulfillRequestWithKeyData(AVAssetResourceLoadingRequest *request, Ar
         auto initData = SharedBuffer::create(Vector<uint8_t> { static_cast<uint8_t*>(initDataBuffer->data()), byteLength });
         player->keyNeeded(initData);
 #if ENABLE(ENCRYPTED_MEDIA)
-        if (!player->shouldContinueAfterKeyNeeded())
```

Comment:
I think this is still racy.

we have fulfillRequestWithKeyData that can be called at any time once the player private has been created, and we will pass the `shouldContinueAfterKeyNeeded` value potentially after it's been first queried.

You've reduced the chance of the race failing because you send it a little bit earlier, but it's still there.

The only way to guarantee that it's not racy is to not create the ResourceRequest until we've provided that information (true or false), 

It will still not work properly if the `webkitneedkey` listener is not added to the HTMLMediaElement until after playback as started.

the existing code is unfortunately quite fragile here.

Reply:
> You've reduced the chance of the race failing because you send it a little bit earlier, but it's still there.
>
> The only way to guarantee that it's not racy is to not create the ResourceRequest until we've provided that information (true or false),

That is exactly what this change does - it calls `setShouldContinueAfterKeyNeeded` from `MediaPlayer::loadWithNextMediaEngine` __before__ setting the media player's source so the information is available before we create the ResourceRequest.

> It will still not work properly if the webkitneedkey listener is not added to the HTMLMediaElement until after playback as started.

True, but if that is a problem it is one we have always had with the legacy API and it apparently isn't an issue.